tools/esdlcmd/esdlcmd_diff.cpp

```diff
@@ -0,0 +1,1139 @@
+                    xml.pad(indent+2).appendf("<%s>\n", item_name);
+                    addMonitoringChildren(xml, esdl, st, indent+4);
+                    xml.pad(indent+2).appendf("</%s>\n", item_name);
+                    xml.pad(indent).appendf("</%s>\n", name);
+                }
+            }
+            else if (streq(elem, "EsdlEnumRef"))
+            {
+                xml.pad(indent).appendf("<%s type='enum'/>\n", name);
+            }
+            else
+            {
+                xml.pad(indent).appendf("<%s unimplemented='%s'/>\n", name, elem);
+            }
+
+        }
+    }
+
+    void createMonitoringTemplate(StringBuffer &xml, IPropertyTree *esdl, const char *method)
+    {
+        xml.append("<ResultMonitoringTemplate>\n");
+        VStringBuffer typeName("%sResponse", method);
+        xml.append("  <").append(typeName).append(">\n");
+        VStringBuffer xpath("*[@name='%s'][1]", typeName.str());
+        IPropertyTree *responseType = esdl->queryPropTree(xpath);
+        if (responseType)
+            addMonitoringChildren(xml, esdl, responseType, 4);
+        xml.append("  </").append(typeName).append(">\n");
+        xml.append("</ResultMonitoringTemplate>");
+    }
+
+    virtual int processCMD()
+    {
+        loadServiceDef();
+
+        StringBuffer xml("<esdl>");
+        Owned<IEsdlDefObjectIterator> structs = cmdHelper.esdlDef->getDependencies( optService.get(), optMethod.get(), ESDLOPTLIST_DELIMITER, 0, nullptr, 0 );
+        cmdHelper.defHelper->toXML(*structs, xml, 0, NULL, 0);
+        xml.append("</esdl>");
+
+        Owned<IPropertyTree> depTree = createPTreeFromXMLString(xml, ipt_ordered);
+        removeEclHidden(*depTree);
+        toXML(depTree, xml.clear());
+
+        StringBuffer monTemplate;
+        createMonitoringTemplate(monTemplate, depTree, optMethod);
+
+        VStringBuffer templatefile("diff_template_%s.xml", optMethod.str());
+        saveAsFile(".", templatefile, monTemplate);
+        return 0;
+    }
+
+    virtual void usage()
+    {
+        puts("\nesdl ecl-diff-template");
+        puts("  Generates a blank Differencing template for the given ESDL based query.\n");
+        puts("Usage:");
+        puts("esdl ecl-diff-template <esdlSourcePath> <serviceName> <methodName> [options]\n" );
+        puts("  <esdlSourcePath> Path to the ESDL Definition file containing the ESDL service definition" );
+        puts("                   to create a ECL result differencing template for." );
+        puts("  <serviceName>    Name of the ESDL Service to generate the template for." );
+        puts("  <methodName>     Name of the ESDL method to generate the template for." );
+
+        EsdlConvertCmd::usage();
+    }
+
+    virtual void loadServiceDef()
+    {
+        cmdHelper.loadDefinition(optSource, optService, 0);
+    }
+
+    void saveAsFile(const char * dir, StringBuffer &outname, const char *text, const char *ext="")
```

Comment:
this looks to be static in nature, and prob useful for other ESDL commands, consider making it visible to other classes?